demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx

```diff
@@ -71,6 +71,11 @@ const createProductMutation = gql`
     }
 `;
 
+declare module "@mui/x-data-grid-pro" {
+    interface ToolbarPropsOverrides {
+        toolbarAction: React.ReactNode;
+    }
+}
```

Comment:
This adds `toolbarAction` to the toolbarProps for all Toolbars in the project, right? I don't think we want this. Is there no other way to do this?

Reply:
This should stay in this scope. I also added this to the [admin generator](https://github.com/vivid-planet/comet/pull/2837/files/4de450fc2b3ad1f89143b8084d85506f2ee9c75a..1d030fcdd2185db73181c527ce147b649a597332#diff-ffbfd5bfa458ffa3572ed41a130f29996a3abf28ea3b90c6c21586c008c8ebb7R118) so typescript is fine with it.

As far as i have seen, there is **NO** other way to do this. Maybe someone know something else :-)

Reply:
I'm not sure about this. Alternatively we could replace the slot altogether:

```ts
slots={{
  toolbar: () => <ProductsGridToolbar {/* Pass props here */}  />,
}}
```

But I think that using slotProps is preferred. Also MUI suggests using module augmentation: https://mui.com/x/react-data-grid/components/#custom-slot-props-with-typescript. If we really don't want module augmentation, we could use as casts: https://mui.com/x/common-concepts/custom-components/#using-additional-props.

> This should stay in this scope.

We should verify that.

Reply:
Unfortuantly it was not scoped, and all augmented ToolbarProps get merged together. Refactored it to use as casts instead.